src/docfx/lib/github/GitHubUserCache.cs

```diff
@@ -56,10 +58,24 @@ private GitHubUserCache(Docset docset, string token)
             var github = new GitHubAccessor(token);
             _getUserByLoginFromGitHub = github.GetUserByLogin;
             _getLoginByCommitFromGitHub = github.GetLoginByCommit;
-            _cachePath = string.IsNullOrEmpty(docset.Config.GitHub.UserCache)
-                ? AppData.GitHubUserCachePath
-                : docset.GetFileRestorePath(docset.Config.GitHub.UserCache);
             _expirationInHours = docset.Config.GitHub.UserCacheExpirationInHours;
+            var path = docset.Config.GitHub.UserCache;
+            if (string.IsNullOrEmpty(path))
+            {
+                _cachePath = AppData.DefaultGitHubUserCachePath;
+                return;
+            }
+            var restorePath = docset.GetFileRestorePath(path);
+            if (HrefUtility.IsHttpHref(path))
```

Comment:
>path [](start = 39, length = 4)

even it's from local file, should be different with your cache file, then why we abandond the cache file?

Reply:
If user explicitly set it a local file, I think mostly it means to read and write to this file.

---
In reply to: [239725092](https://github.com/dotnet/docfx/pull/3740#discussion_r239725092) [](ancestors = 239725092)